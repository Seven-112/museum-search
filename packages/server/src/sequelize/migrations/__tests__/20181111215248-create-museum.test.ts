// tslint:disable-next-line: no-var-requires
const createMuseums = require("../20181111215248-create-museum");

const mockCreateTable = jest.fn();
const mockDropTable = jest.fn();

const mockQueryInterface = {
  createTable: mockCreateTable,
  dropTable: mockDropTable
};

// Mock the sequelize types as strings so they show up as type names instead of "[Function]" in snapshots.
const MOCK_SEQUELIZE_TYPES = {
  DATE: "DATE",
  DOUBLE: "DOUBLE",
  INTEGER: "INTEGER",
  STRING: "STRING"
};

describe("create museums table migration", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("Creates the museums table.", () => {
    createMuseums.up(mockQueryInterface, MOCK_SEQUELIZE_TYPES);
    expect(mockCreateTable).toHaveBeenCalledTimes(1);
    // Migrations shouldn't change, so snapshots can enforce that.
    expect(mockCreateTable.mock.calls[0]).toMatchSnapshot(
      "create museums table migration."
    );
  });

  it("Drops the museums table.", () => {
    createMuseums.down(mockQueryInterface);
    expect(mockDropTable).toHaveBeenCalledTimes(1);
    expect(mockDropTable).lastCalledWith("Museums");
  });
});
