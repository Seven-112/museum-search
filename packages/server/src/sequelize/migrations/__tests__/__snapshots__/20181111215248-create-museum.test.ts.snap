// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create museums table migration Creates the museums table.: create museums table migration. 1`] = `
Array [
  "Museums",
  Object {
    "alternateName": Object {
      "type": "STRING",
    },
    "city": Object {
      "type": "STRING",
    },
    "createdAt": Object {
      "allowNull": false,
      "type": "DATE",
    },
    "id": Object {
      "allowNull": false,
      "autoIncrement": true,
      "primaryKey": true,
      "type": "INTEGER",
    },
    "institutionName": Object {
      "type": "STRING",
    },
    "latitude": Object {
      "type": "DOUBLE",
    },
    "legalName": Object {
      "type": "STRING",
    },
    "longitude": Object {
      "type": "DOUBLE",
    },
    "museumType": Object {
      "type": "STRING",
    },
    "name": Object {
      "type": "STRING",
    },
    "phoneNumber": Object {
      "type": "STRING",
    },
    "state": Object {
      "type": "STRING",
    },
    "streetAddress": Object {
      "type": "STRING",
    },
    "updatedAt": Object {
      "allowNull": false,
      "type": "DATE",
    },
    "zipCode": Object {
      "type": "STRING",
    },
  },
]
`;
